ci/pipelines/geode-build/jinja.template.yml

```diff
@@ -186,7 +186,7 @@ resources:
     driver: gcs
     initial_version: {{ metadata.initial_version }}
     json_key: ((!concourse-gcp-key))
-    key: semvers/((pipeline-prefix))((geode-build-branch))/version
+    key: semvers/((pipeline-prefix))((geode-build-branch))/number
```

Comment:
I'm curious about this change, but glad the issue is getting fixed

Reply:
This change is to streamline the geode-build-version to use a common file name everywhere. We had version and number being created as part of that resource previously. And even though we define it as version gcp creates a semver resource with the name number.

Reply:
that makes sense